graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/searchtypes/pivot/ESPivotBucketSpecHandler.java

```diff
@@ -46,12 +49,24 @@ protected void record(ESGeneratedQueryContext queryContext, Pivot pivot, PivotSp
         aggTypes(queryContext, pivot).record(spec, name, aggregationClass);
     }
 
-    protected List<BucketOrder> orderListForPivot(Pivot pivot, ESGeneratedQueryContext esGeneratedQueryContext, BucketOrder defaultOrder) {
+    public record SortOrders(List<BucketOrder> orders, List<AggregationBuilder> subAggregations) {}
+
+    protected SortOrders orderListForPivot(Pivot pivot, ESGeneratedQueryContext esGeneratedQueryContext, BucketOrder defaultOrder, Query query) {
```

Comment:
What do you think about some unit tests checking this method?
It may be nice not only for testing, but also documenting different cases and expected outcome for them.